module_utils/network_lsr/argument_validator.py

```diff
@@ -246,6 +246,51 @@ def _validate_string_min_length(self, value):
             return True
 
 
+class ArgValidatorRouteTable(ArgValidator):
+    def __init__(
+        self,
+        name,
+        required=False,
+        default_value=None,
+    ):
+        ArgValidator.__init__(self, name, required, default_value)
+
+    def _validate_impl(self, value, name):
+        v = None
+        val_min = None
+        val_max = None
+        try:
+            if isinstance(value, bool):
+                # bool can (probably) be converted to self.numeric_type,
+                # but here we don't want to accept a boolean value.
+                pass
+            elif isinstance(value, int):
+                val_min = 1
+                val_max = UINT32_MAX
+                v = int(value)
+            elif isinstance(value, Util.STRING_TYPE):
+                v = value
```

Comment:
I think you should try to convert the string to number. If it's a number, then use the number and don't interpret that as an alias.

Reply:
Updated.
```python
            elif isinstance(value, Util.STRING_TYPE):
                if value.isnumeric():
                    v = int(value)
                else:
                    v = value
```

Reply:
`isnumeric()` seems not the right thing here... it checks whether all characters are (individually) numeric characters. OK, that might amount to in fact being the same, but still.

What I mean is that in my mind there is no guarantee that from `value.isnumeric()` always follows that a conversion `int(value)` is going to succeed. These are not the same things. At least, it wouldn't hold for `value="-1"` -- which you might say, is a case we don't want to handle here (but I think we do. We should also not allow "-1" as a valid alias).

I think the best way to check whether a string can be converted to an int is by just trying to do it.
```
    try:
        v = int(value)
    except:
        v = value
```

Reply:
I changed.